server-spi/src/main/java/org/keycloak/storage/user/UserQueryProvider.java

```diff
@@ -214,7 +214,7 @@ default int getUsersCount(RealmModel realm, boolean includeServiceAccount) {
      * @param firstResult
      * @param maxResults
      * @return
-     * @deprecated Use {@link #searchForUserStream(String, RealmModel, int, int) searchForUserStream} instead.
+     * @deprecated Use {@link #searchForUserStream(String, RealmModel, Integer, Integer) searchForUserStream} instead.
      */
     @Deprecated
     List<UserModel> searchForUser(String search, RealmModel realm, int firstResult, int maxResults);
```

Comment:
This is good question, I am not sure. @hmlnarik wanted to leave `BackwardsCompatibilityUserStorage` as is to implement the old variant. Not sure about the others. There are more providers in testsuite that implement old variants. If we decide to make all except for `BackwardsCompatibilityUserStorage` implement `Streams` variants I can make it part of Cleanup PR. @hmlnarik @sguilhen WDYT?

Reply:
At this moment, either option is open. It would be better to follow @vramik's suggestion if that does not represent a big amount of work at this point. In the opposite case, we can postpone this for either cleanup or even later when removing the interaces.

Reply:
I will make it part of https://issues.redhat.com/browse/KEYCLOAK-15524. Does it work for you @vramik @hmlnarik?

Reply:
It works for me.